libraries/AP_HAL/hwdef/scripts/hwdef.py

```diff
@@ -17,7 +17,10 @@ def __init__(self, hwdef, includer):
 
 
 class HWDef:
-    def __init__(self, quiet=False, outdir=None, hwdef=[]):
+    def __init__(self, quiet=False, outdir=None, hwdef : list = None):
+        if hwdef is None:
```

Comment:
All falsey values (esp `False`) are just as dangerous as None.
```suggestion
        if not hwdef:
```

I doubt that Python type checkers `mypy` or `ty` or `pyrefly` would love `hwdef: list = None` as they expect to see a constraining set of types `hwdef: list | None = None` instead.  `hwdef` is a list or `None` or they raise a type error.

Reply:
If someone passes in `False` then I want the thing to creatively die, *not* "fix" the problem.

I was wondering about that requirement for allowing None especially.

I'm rapidly coming to the conclusion that I *absolutely should not* have added this typing information as part of this PR - it was completely optional for me to do so and seems to be unnecessarily blocking my fixes for the actual bugs that were evident.

Reply:
This is a non sequitur, if you want to build a particular version of ArduPilot you have to use the associated Python bits. You can't source them from an earlier release.

We are stuck on 3.9 as Cygwin still needs it.